azure-iot-device/azure/iot/device/iothub/abstract_clients.py

```diff
@@ -30,6 +27,28 @@
 # SymmetricKeyAuthenticationProvider was intended to be part of an Edge scenario or not.
 
 
+def _validate_kwargs(**kwargs):
+    """Helper function to validate user provided kwargs.
+    Raises TypeError if an invalid option has been provided"""
+    valid_kwargs = ["product_info", "websockets", "cipher", "server_verification_cert"]
+
+    for kwarg in kwargs:
+        if kwarg not in valid_kwargs:
+            raise TypeError("Got an unexpected keyword argument '{}'".format(kwarg))
+
+
+def _get_pipeline_config_kwargs(**kwargs):
```

Comment:
>_get_pipeline_config_kwargs [](start = 4, length = 27)

I am not sure what is this doing

Reply:
This is getting the subset of **kwargs that is relevant to a IoTHubPipelineConfig. Not all **kwargs are passed to IoTHubPipelineConfig - specifically, 'server_verification_cert'

---
In reply to: [374839219](https://github.com/Azure/azure-iot-sdk-python/pull/434#discussion_r374839219) [](ancestors = 374839219)